site/content/en/docs/Guides/access-api.md

```diff
@@ -134,6 +134,130 @@ kubectl logs create-gs-6wz86-7qsm5  --namespace agones-system
+and can execute a callback with the relevant object as an argument.
+
+This can be very useful for building event based systems against the Kubernetes API.
+
+A Lister is the mechanism for querying Kubernetes object's against the client side in-memory cache.
+Since the Lister stores objects in an in-memory cache, queries against a come at practically no cost.
+
+Of course, Agones itself also uses Informers and Listers in its codebase.
+
+### Example
+
+The following is an example of Informers and Listers,
+that show the GameServer's name & status & IPs in the Kubernetes cluster.
+
+```go
+package main
+
+import (
+	"log"
+	"time"
+
+	"agones.dev/agones/pkg/client/clientset/versioned"
+	"agones.dev/agones/pkg/client/informers/externalversions"
+	"agones.dev/agones/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/rest"
+)
+
+func main() {
+	config, err := rest.InClusterConfig()
+	logger := runtime.NewLoggerWithSource("main")
+	if err != nil {
+		logger.WithError(err).Fatal("Could not create in cluster config")
+	}
+	kubeClient, err := kubernetes.NewForConfig(config)
+	agonesClient, err := versioned.NewForConfig(config)
+	if err != nil {
+		logger.WithError(err).Fatal("Could not create the agones api clientset")
+	}
+
+	// Create InformerFactory which create the informer
+	informerFactory := informers.NewSharedInformerFactory(kubeClient, time.Second*30)
+	agonesInformerFactory := externalversions.NewSharedInformerFactory(agonesClient, time.Second*30)
+
+	// Create Pod informer by informerFactory
+	podInformer := informerFactory.Core().V1().Pods()
+
+	// Create GameServer informer by informerFactory
+	gameServers := agonesInformerFactory.Agones().V1().GameServers()
+	gsInformer := gameServers.Informer()
+
+	// Add EventHandler to informer
+	// When the object's event happens, the function will be called
+	// For example, when the pod is added, 'AddFunc' will be called and put out the "Pod Added"
+	podInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    func(new interface{}) { logger.Infof("Pod Added") },
+		UpdateFunc: func(old, new interface{}) { logger.Infof("Pod Updated") },
+		DeleteFunc: func(old interface{}) { logger.Infof("Pod Deleted") },
+	})
+	gsInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    func(new interface{}) { logger.Infof("GameServer Added") },
+		UpdateFunc: func(old, new interface{}) { logger.Infof("GameServer Updated") },
+		DeleteFunc: func(old interface{}) { logger.Infof("GameServer Deleted") },
+	})
+
+	// Start Go routines for informer
+	informerFactory.Start(wait.NeverStop)
+	agonesInformerFactory.Start(wait.NeverStop)
+	// Wait until finish caching with List API
+	informerFactory.WaitForCacheSync(wait.NeverStop)
+	agonesInformerFactory.WaitForCacheSync(wait.NeverStop)
```

Comment:
🤔 do we want to use wait.NeverStop here, or do we want to drive people towards what they probably should do - which is have a `context.Context` (maybe cancellable? Might not matter, could just use [context.Background](https://pkg.go.dev/context#Background)), and pass in ctx.Done() into each of these?

Since it's likely better practice to use a `Context`. WDYT?

Also /cc @roberthbailey in case he has opinions.

Reply:
changed using `context.Background`.